File: problem1221.py
class Solution:
    def balancedStringSplit(self, s: str) -> int:
        count=0
        result=0
        for c in s:
            count=count+1 if c=='L' else count - 1 
            if count==0:
                result+=1
                
        return result
        
#Time=O(N)
#space:O(1)

File: problem1221.cs
public class Solution {
    public int BalancedStringSplit(string s) {
        int balanceCount=0;
        int count=0;
        
        foreach(char c in s)
        {
            if(c=='R')
                balanceCount++;
            else
                balanceCount--;
            
            if(balanceCount==0)
                count++;
        }
        return count;
    }
}

//time: O(N)
//space: O(1)